Cython/Compiler/Parsing.py

```diff
@@ -3390,13 +3390,10 @@ def p_c_func_or_var_declaration(s, pos, ctx):
     else:
         is_const_method = 0
     if s.sy == '->':
```

Comment:
We usually just define defaults for optional syntax parts upfront in the parser.
```suggestion
    return_type_annotation = None
    if s.sy == '->':
```